app/services/generate_counter_metrics.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+# Creates counter metrics for a single tenant.
+class GenerateCounterMetrics
+  def self.generate_counter_metrics(ids: :all, limit: :all)
+    fsq = "has_model_ssim: (#{Bulkrax.curation_concerns.join(' OR ')})"
+    fsq += " AND id:(\"" + Array.wrap(ids).join('" OR "') + "\")" if ids.present? && ids != :all
+    message = if ids == :all || ids.blank?
+                "Creating test data for #{limit} randomly selected works"
```

Comment:
I'm a bit confused by the message.  In particular why we're concerned about ids but then the message is about limits.

How about:

```ruby
message = "#{self.class}.generate_counter_metrics data for IDs = #{ids.inspect} with Limit = #{limit.inspect}."
```

Less conditionals and we're not worried about the interplay of `:ids` and `:limits` (which are mostly orthogonal in nature)